File: leetcode/easy/Array/136_Single Number.java
class Solution {
    public int singleNumber(int[] nums) {
        
        Map<Integer, Integer> map = new HashMap<>();
        
        for(int i = 0; i < nums.length; i++){
            map.put(nums[i], map.getOrDefault(nums[i] , 0) + 1);            
        }
        
        return getKey(map,1);
    }
    
    public <K, V> K getKey(Map<K, V> map, V value) {
        for (K key : map.keySet()) {
            if (value.equals(map.get(key))) {
                return key;
            }
        }
        return null;
    }
}

File: leetcode/easy/Bit Manipulation/cpp/136_Single Number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        
        if(nums.size() == 1) return nums[0]; 
        
        sort(nums.begin(), nums.end());
        
        for(int i = 0; i < nums.size(); i+=2){
            
            if(nums[i] != nums[i+1]){
                return nums[i];
            }
        }
        
        return -1;
    }
};